### ISSUE TITLE
How to fix flex-gap issue in the best way? Mantine with Next.js

### ISSUE THREAD

**Issue by totaldekadens** [mantinedev/mantine#2804]
### What package has an issue

@mantine/hooks

### Describe the bug

Hi! 

I have some problem with flex-gap in older browsers. I know it's current issue and you mention it in your site. But do you have a solution to prevent the flex-gap not working in older browsers with Next.js?  

### What version of @mantine/hooks page do you have in package.json?

"^5.4.1"

**Comment by rtivital** [comment]
No, there is no solution for this, if you want to support older browsers, then you should not use components that rely on this property

**Comment by totaldekadens** [comment]
Thanks for your quick response :)

**Comment by 7iomka** [comment]
Flex Gap is not supported for Safari < 14.1, but used in
  - Carousel: `indicators: {gap: 8}`
  - Group: `root: { gap: theme.fn.size({ size: spacing, sizes: theme.spacing }) }`
  - Stack: `root: { gap: theme.fn.size({ size: spacing, sizes: theme.spacing }) }`
  - Tabs (TabList): `tabsList: {... if (variant === 'pills') { return { gap: theme.spacing.sm / 2, }; } ... }`
  - Calendar (CalendarBase): `calendarBase: {gap: theme.spacing.md}`

**Note**: Group and Stack components is a base for **_most_** Mantine components.

So, workaround is a pretty simple for generic fix:
In mantine provider inside `components` put code:
```ts
Group: {
  styles: (theme: MantineTheme, { spacing }: GroupStylesParams) => {
    return {
      root: {
        gap: 0,
        marginRight: -theme.fn.size({ size: spacing, sizes: theme.spacing }),
        marginBottom: -theme.fn.size({ size: spacing, sizes: theme.spacing }),
        '> *': {
          marginRight: theme.fn.size({ size: spacing, sizes: theme.spacing }),
          marginBottom: theme.fn.size({ size: spacing, sizes: theme.spacing }),
        },
      },
    };
  },
},
Stack: {
  styles: (theme: MantineTheme, { spacing }: StackStylesParams) => {
    return {
      root: {
        gap: 0,
        marginBottom: -theme.fn.size({ size: spacing, sizes: theme.spacing }),
        '> *': {
          marginBottom: theme.fn.size({ size: spacing, sizes: theme.spacing }),
        },
      },
    };
  },
},
```

For rest of components you can modify settings in similar way.
Optionally, you can also use `modernizr` which will add the className like `no-flexgap` to the html element, and you will be able to use this workaround only when this className is present, in other cases native gap will be used.
